src/LibraryManager.Vsix/UI/Controls/TargetLocation.xaml.cs

```diff
@@ -26,14 +26,19 @@ public partial class TargetLocation : INotifyPropertyChanged
             nameof(Text), typeof(string), typeof(TargetLocation), new PropertyMetadata(default(string)));
 
         private string _text;
+        private string _lastTargetLocation; 
+        private string _baseFolder;
 
         public TargetLocation()
         {
             InitializeComponent();
 
             // Pre populate textBox with folder name
             TargetLocationSearchTextBox.Text = InstallationFolder.DestinationFolder;
-
+            _baseFolder = InstallationFolder.DestinationFolder;
+            _lastTargetLocation = InstallationFolder.DestinationFolder;
+            MutualPropertyChange.Instance.TargetLibrary = string.Empty;
```

Comment:
>Instance [](start = 33, length = 8)

Let's not use a singleton here. You can just create an instance of the class for this dialog. It can be part of the InstallDialogViewModel.

Reply:
Need to set the event listener after the targetLcation get loaded because the initialization of target location takes place prior to the initialization of InstallDialogViewModel

---
In reply to: [200771611](https://github.com/aspnet/LibraryManager/pull/285#discussion_r200771611) [](ancestors = 200771611)